packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx

```diff
@@ -357,7 +362,22 @@ export class AtomicSearchInterface {
     });
   }
 
+  private fixFacetDependencies() {
+    if (getIncorrectFacetEnableStates(this.engine!, this.store.state).length) {
+      this.ignoreUrlManagerStateChanges = true;
+      fixFacetDependencies(this.engine!, this.store);
+      this.ignoreUrlManagerStateChanges = false;
+    }
+  }
+
   private updateHash() {
+    if (this.ignoreUrlManagerStateChanges) {
+      return;
+    }
+    this.fixFacetDependencies();
```

Comment:
Notice here that we don't `return` when facet dependencies are invalid, we just activate the `ignoreUrlManagerStateChanges` flag while fixing dependencies and then we continue.

The reason for this is because fixing invalid facet dependencies doesn't always call `updateHash` a second time. For instance, when we enable facets.